Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -923,24 +939,27 @@ namespace AZ
         }
     }
 
-    void ComponentApplication::MergeSettingsToRegistry(SettingsRegistryInterface& registry)
+    void ComponentApplication::MergeSharedSettings(
+        SettingsRegistryInterface& registry,
+        const AZ::SettingsRegistryInterface::Specializations& specializations,
+        AZStd::vector<char>& scratchBuffer)
     {
-        SettingsRegistryInterface::Specializations specializations;
-        SetSettingsRegistrySpecializations(specializations);
+        if constexpr (GetDevelopmentSettingsOverrides() == DevelopmentSettingsOverrides::CommandLineProjectAndUser)
+        {
+            // In development builds apply the o3de registry and the command line to allow early overrides. This will
+            // allow developers to override things like default paths or Asset Processor connection settings. Any additional
+            // values will be replaced by later loads, so this step will happen again at the end of loading.
+            SettingsRegistryMergeUtils::MergeSettingsToRegistry_O3deUserRegistry(
+                registry, AZ_TRAIT_OS_PLATFORM_CODENAME, specializations, &scratchBuffer);
+            SettingsRegistryMergeUtils::MergeSettingsToRegistry_CommandLine(registry, m_commandLine, false);
+            // Project User Registry is merged after the command line here to allow make sure the any command line override of the project
+            // path is used for merging the project's user registry
+            SettingsRegistryMergeUtils::MergeSettingsToRegistry_ProjectUserRegistry(
+                registry, AZ_TRAIT_OS_PLATFORM_CODENAME, specializations, &scratchBuffer);
+            SettingsRegistryMergeUtils::MergeSettingsToRegistry_CommandLine(registry, m_commandLine, false);
+            SettingsRegistryMergeUtils::MergeSettingsToRegistry_AddRuntimeFilePaths(registry);
+        }
```

Comment:
We should provide similar logic for making sure the command line is merged as we do in MergeUserSettings, when we the override settings are `CommandLineOnly` or `CommandLineAndProject`

We can check if the `GetDevelopmentSettingOverrides()` enum is greater than or equal to the `CommandLineOnly` option to enter the block below and then add an inner check to merge the o3de user registry and project user registry if the value is `CommandLineProjectAndUser`
```suggestion
        if constexpr (GetDevelopmentSettingsOverrides() >= DevelopmentSettingsOverrides::CommandLineOnly)
        {
            // In development builds apply the o3de registry and the command line to allow early overrides. This will
            // allow developers to override things like default paths or Asset Processor connection settings. Any additional
            // values will be replaced by later loads, so this step will happen again at the end of loading.
            if constexpr (GetDevelopmentSettingsOverrides() == DevelopmentSettingsOverrides::CommandLineProjectAndUser)
            {
                SettingsRegistryMergeUtils::MergeSettingsToRegistry_O3deUserRegistry(
                    registry, AZ_TRAIT_OS_PLATFORM_CODENAME, specializations, &scratchBuffer);
                SettingsRegistryMergeUtils::MergeSettingsToRegistry_CommandLine(registry, m_commandLine, false);
            }
                // Project User Registry is merged after the command line here to allow make sure the any command line override of the project
                // path is used for merging the project's user registry
                if constexpr (GetDevelopmentSettingsOverrides() == DevelopmentSettingsOverrides::CommandLineProjectAndUser)
                {
                    SettingsRegistryMergeUtils::MergeSettingsToRegistry_ProjectUserRegistry(
                        registry, AZ_TRAIT_OS_PLATFORM_CODENAME, specializations, &scratchBuffer);
                }
                
            // Make sure the command line is merged at least once, before updating the runtime filepaths
            SettingsRegistryMergeUtils::MergeSettingsToRegistry_CommandLine(registry, m_commandLine, false);
            SettingsRegistryMergeUtils::MergeSettingsToRegistry_AddRuntimeFilePaths(registry);
        }

```